### PR TITLE
Add missing methods for tms integration and fix issues about entity consumer services cleanup

### PR DESCRIPTION
**Pull request by mathieucarbou**
This is a for for some tests and also, more importantly, I completely get rid of the weak references to better handle the creation and destruction of entity services holding states (client descriptors).

this was causing some management call issues sometimes.

### CODE REVIEW

**Comment by mathieucarbou** [comment]
test this please

File: management/monitoring-service/src/main/java/org/terracotta/management/service/monitoring/DefaultClientMonitoringService.java [hunk 1]
```diff
@@ -98,6 +99,24 @@ public void answerManagementCall(ClientDescriptor caller, String managementCallI
     eventService.fireManagementCallAnswer(managementCallIdentifier, contextualReturn);
   }
 
+  @Override
+  public void onFetch(long consumerId, ClientDescriptor clientDescriptor) {
```
Comment (gibsong) [inline management/monitoring-service/src/main/java/org/terracotta/management/service/monitoring/DefaultClientMonitoringService.java]: Why isn't the onFetch(...) method implemented?

File: management/monitoring-service/src/main/java/org/terracotta/management/service/monitoring/DefaultConsumerManagementRegistry.java [hunk 2]
```diff
@@ -96,4 +90,21 @@ public boolean pushServerEntityNotification(Object managedObjectSource, String t
     return false;
   }
 
+  @Override
+  public void onFetch(long consumerId, ClientDescriptor clientDescriptor) {
```
Comment (gibsong) [inline management/monitoring-service/src/main/java/org/terracotta/management/service/monitoring/DefaultConsumerManagementRegistry.java]: Why isn't onFetch(...) implemented?

File: management/monitoring-service/src/main/java/org/terracotta/management/service/monitoring/DefaultConsumerManagementRegistry.java [hunk 3]
```diff
@@ -96,4 +90,21 @@ public boolean pushServerEntityNotification(Object managedObjectSource, String t
...
+  }
+
+  @Override
+  public void onUnfetch(long consumerId, ClientDescriptor clientDescriptor) {
```
Comment (gibsong) [inline management/monitoring-service/src/main/java/org/terracotta/management/service/monitoring/DefaultConsumerManagementRegistry.java]: Why isn't onUnFetch(...) implemented?

**Comment by mathieucarbou** [comment]
test this please

**Comment by mathieucarbou** [comment]
@GaryWKeim @gibsong @anthonydahanne : someone can merge please ?

**Pull request merged**